main/src/main/scala/sbt/Main.scala

```diff
@@ -153,9 +153,13 @@ private[sbt] object xMain {
           if (ITerminal.get.withRawInput(System.in.read) == 'n'.toInt) Some(Exit(1))
           else None
         (None, exit)
-      case _: ServerAlreadyBootingException =>
+      case e: ServerAlreadyBootingException =>
         if (SysProp.forceServerStart) (None, None)
-        else (None, Some(Exit(2)))
+        else {
+          println("Boot server failed to create socket")
+          e.printStackTrace()
```

Comment:
Could we refactor line 149~150 out to a in-function function so we can use the same error message please?

Reply:
I'd be happy to but I'm not quote sure what you are wanting. So a couple of questions:
1. What message do you want to re-use?
2. What scope do you want the function defined in?

If you are think of reusing the above error message, this case is a little different in that we are not going to take input from the user. But if that what you mean, just the first two bit of print outs, then that fine and I can do that. 

Also if you have another place that models the pattern, just point me to it and I can adapt it to this case.

Reply:
> But if that what you mean, just the first two bit of print outs, then that fine and I can do that.

Yea. That's what I was thinking. People search for error messages on GitHub issues and Stackoverflow etc so I want to make sure we have consistent error message for the same class of error.

So:

> 1. What message do you want to re-use?
```scala
        println("sbt thinks that server is already booting because of this exception:")
        e.printStackTrace()
```

and 

> 2. What scope do you want the function defined in?

I think the narrowest scope would be a nested method.

Reply:
Done, let me know if it match what you were looking for.

Reply:
Yes. The change LGTM.